packages/formspree-core/test/utils.test.ts

```diff
@@ -0,0 +1,116 @@
+import { appendExtraData } from '../src/utils';
+
+describe('appendExtraData', () => {
+  describe('given a plain object input', () => {
+    type TestCase = {
+      name: string;
+      input: {
+        formData: Record<string, string>;
+        prop: string;
+        value: string;
+      };
+      expected: Record<string, string>;
+    };
+
+    const testCases: TestCase[] = [
+      {
+        name: 'empty',
+        input: {
+          formData: {},
+          prop: 'foo',
+          value: 'bar',
+        },
+        expected: { foo: 'bar' },
+      },
+      {
+        name: 'with some existing values',
+        input: {
+          formData: { a: '1', b: '2' },
+          prop: 'foo',
+          value: 'bar',
+        },
+        expected: { a: '1', b: '2', foo: 'bar' },
+      },
+      {
+        name: 'with an existing value for the same prop',
+        input: {
+          formData: { foo: '(existing)' },
+          prop: 'foo',
+          value: 'bar',
+        },
+        expected: { foo: 'bar' },
+      },
+    ];
+
+    test.each(testCases)('$name', ({ input, expected }) => {
+      const { formData, prop, value } = input;
+      appendExtraData(formData, prop, value);
+      expect(formData).toEqual(expected);
+    });
+  });
+
+  describe('given a FormData input', () => {
+    type TestCase = {
+      name: string;
+      input: {
+        formData: FormData;
+        prop: string;
+        value: string;
+      };
+      expectedEntries: [string, string][];
+    };
+
+    const testCases: TestCase[] = [
+      {
+        name: 'empty',
+        input: {
+          formData: createFormData({}),
+          prop: 'foo',
+          value: 'bar',
+        },
+        expectedEntries: [['foo', 'bar']],
+      },
+      {
+        name: 'with some existing values',
+        input: {
+          formData: createFormData({ a: '1', b: '2' }),
+          prop: 'foo',
+          value: 'bar',
+        },
+        expectedEntries: [
+          ['a', '1'],
+          ['b', '2'],
+          ['foo', 'bar'],
+        ],
+      },
+      {
+        name: 'with an existing value for the same prop',
+        input: {
+          formData: createFormData({ foo: '(existing)' }),
+          prop: 'foo',
+          value: 'bar',
+        },
+        expectedEntries: [
+          ['foo', '(existing)'],
+          ['foo', 'bar'],
+        ],
```

Comment:
This is how the current implementation works. It probably is unintentional, since it'll use different semantic when the `formData` is a plain object versus when the `formData` is a type of `FormData`.

Would it make sense to change the implementation to use `formData.set` instead of `formData.append`, so the behavior is mirrored between the two input types?

Reply:
Using `appendExtraData` to override existing data is an unexpected edgecase. The method is intended as an alternative to hidden fields in HTML. Probably if there's a collision it should print a warning to the console and then perform best effort merging. In the formdata case, duplicate fields makes sense, but in the POJO case, best effort is overwrite.